README_EN.md

```diff
@@ -0,0 +1,233 @@
+    }],
+    "deploy_time": "Sun, 09 Oct 2016 07:43:47 GMT",
+}
+```
+
+### Configure the resource path
+
+```Swift
+  RXRConfig.setRoutesResourcePath("rexxar")
+```
+
+Using Rexxar to develop mobile application, we usually put a copy of resource files in the application's release package. This copy of resource files in app's package will reduce user's download of resource files and optimise the user's experience of application.
+
+If we configure the resource path in app's package of Rexxar, the type of folder including the resource files must be `folder reference` which is blue folder icon in Xcode, rather than yellow folder icon of `group` type.
+
+### Configure the cache path
+
+```Swift
+  RXRConfig.setRoutesCachePath("com.douban.RexxarDemo.rexxar")
+```
+
+The cache can also reduce user's download of resource files and optimise the user's experience of application.
+
+The cache will be created after Rexxar Container gets the update of routes map file. 
+
+The routes map file decides the update of resouces files. Application visits the routes map api periodically. If it gets the update of routes map, Rexxar Container will download the update of resource files and save the copy in the cache path. That's the way Rexxar supports `hot deploy`. You don't need release your application via App Store. In stead of that, Rexxar update the pages after the update of routes map file.
+
+This is the way to call to update routes map file:
+
+```Swift
+  RXRViewController.updateRouteFiles(completion: nil)
+```
+
+### Use `RXRViewController`
+
+You can use `RXRViewController` directly as your hybrid container. Or you can inherit `RXRViewController` to implement your own Rexxar Container. In the Demo, we implement `FullRXRViewController` inheriting from `RXRViewController`.
+
+To initialize a RXRViewController, you just need a valid url. This url should exist in the routes map file. Every url represents a page in app. Rexxar Container search the page's resource files including html, css, javascript via the url in the routes map file.
+
+```Swift
+     let controller = RXRViewController(URI: uri)
+     let titleWidget = RXRNavTitleWidget()
+     let alertDialogWidget = RXRAlertDialogWidget()
+     controller.activities = [titleWidget, alertDialogWidget]
+     navigationController?.pushViewController(controller, animated: true)
+```
+
+
+## Customize your own Rexxar Container
+
+First, you inherit `RXRViewController` to implement your own Rexxar Container.
+
+Then, you use the three interfaces provided by Rexxar to make your customization easier.
+
+We will introduce how to use the three interfaces in the following sections.
+
+### Create your own RXRWidget
+
+If you want to implement a native UI feature which can be used by web, for example, displaying a toast or adding pull to refresh UI widget etc, you can inherit `RXRWidget` and implement the three methods: `canPerformWithURL:`, `prepareWithURL:`, `performWithContoller:`.
```

Comment:
A `RXRWidget` provides three methods: `canPerformWithURL:`, `prepareWithURL:`, `performWithContoller:`. Override these three methods to let Rexaar know how should a URL trigger a native UI feature. （? Am I getting this right?)